### DISCUSSION TITLE
Kubernetes Executor pods not deleting on Kubernetes cluster with linkerd container injected

### DISCUSSION THREAD

**Discussion by KGNickl** [apache/airflow#16033] [General]
This is happening to the upgrade-db pod after starting up and also happening to pods launched by the kubernetes executioner when they complete. The pods go into a "NotReady" state. The kubernetes cluster has linkerd container auto injected into all pods and I believe this is causing this and I'm not sure if there is a best way to handle this in airflow so the pods do not hang around because of the linkerd container in the pod.

I installed airflow in kubernetes (azure AKS) using the repo https://airflow-helm.github.io/charts and airflow-stable/airflow with version 8.1.3, so I have Airflow v2.0.1 installed. I have it setup using external postgres database and using the kubernetes executor.

I set the following (Values.airflow.config. ):

AIRFLOW__KUBERNETES__DELETE_WORKER_PODS: "true" 
AIRFLOW__KUBERNETES__DELETE_WORKER_PODS_ON_FAILURE: "true"

NAME                                                         READY   STATUS     RESTARTS   AGE
airflow2-scheduler-5fdb9b56b5-zqzgk                          2/2     Running    0          36m
airflow2-upgrade-db-8s8f2                                    1/2     NotReady   0          36m
airflow2-web-74fc74d87-jgrr9                                 2/2     Running    0          36m
demodagechoairflow.1a9943de55204fa9ac144e0a7d65b58a          1/2     NotReady   0          29m
demodagechoairflow.e48f978b8287477f856d794d78b122a0          1/2     NotReady   0          30m
demodagechoairflow.f7b4fe331b734acd8dc837245339e6b3          1/2     NotReady   0          29m
demodagechoapache.64c97c0bac434212b3dec81e0a37b85d           1/2     NotReady   0          32m
demodagechoapache.bf4d6389610d493c8c26b968a4c4877e           1/2     NotReady   0          32m
demodagechoapache.c30f5c97fb2d43f592a5a3073a0f26d2           1/2     NotReady   0          31m
demodagechodemo.1b41193593a246feb9d936df6f815ae4             1/2     NotReady   0          28m
demodagechodemo.84da7877118943789a5c50079145cb54             1/2     NotReady   0          27m
demodagechodemo.9c1749f40d594de28eee78f89d56a581             1/2     NotReady   0          27m
testazuredatalakedagtask1.1943abb5ce8c4d4abdfcc06765204eff   1/2     NotReady   0          6m22s
testazuredatalakedagtask1.ac7d263ca4124ab6896a5310146a9370   1/2     NotReady   0          17m

**Comment by jedcunningham** [comment]
Unfortunately KubernetesExecutor doesn't support arbitrary sidecars, so you'd want to use CeleryExecutor instead.

As for the chart, annotating the job so it doesn't get the sidecar is one solution. Not sure about the old stable chart, but the community chart will get support for adding custom annotations to the database migration job in the next chart release (#16058).
